### ISSUE TITLE
install error

### ISSUE THREAD

**Issue by jokerbo1** [sevlyar/go-daemon#28]
Hello ,i am exec go get url  ,then prompt follow message:
github.com/sevlyar/go-daemon
../../../go/src/github.com/sevlyar/go-daemon/lock_file.go:53:9: undefined: lockFile
../../../go/src/github.com/sevlyar/go-daemon/lock_file.go:58:9: undefined: unlockFile
../../../go/src/github.com/sevlyar/go-daemon/lock_file.go:123:9: undefined: getFdName

please ,help me ,thanks  ,
runtime environment 
macos osx
go 1.9

**Comment by sevlyar** [comment]
@359274651, thank you for the report. Could you provide the next information:
```
go version
go env
uname -a
```
And try to execute the next command: `go get -v -x -v github.com/sevlyar/go-daemon`

**Comment by b-ryan** [comment]
Hey I am getting this same error, but when I try to build for mac:

```
 $  GOOS=darwin GOARCH=amd64 go build 
# github.com/sevlyar/go-daemon
../github.com/sevlyar/go-daemon/lock_file.go:53:9: undefined: lockFile
../github.com/sevlyar/go-daemon/lock_file.go:58:9: undefined: unlockFile
../github.com/sevlyar/go-daemon/lock_file.go:123:9: undefined: getFdName
```

I am using Ubuntu 16.04 & the output of the commands you posted:

```
 $  go version
go version go1.9.3 linux/amd64
 $  go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/buck/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build644192553=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
 $  uname -a
Linux buck-Kudu 4.13.0-32-generic #35~16.04.1-Ubuntu SMP Thu Jan 25 10:13:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by b-ryan** [comment]
I have tracked this down to the `import "C"` in `lock_file_darwin.go`. My understanding of the whole situation is limited. Seems like the options are for there to be a different way to get the name of file descriptors or I just need to compile on mac?

**Comment by sevlyar** [comment]
@b-ryan cgo is turned off by default when cross-compiling. So that you should set `CGO_ENABLED` environment variable to `1` in that case: `GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 go build`

**Comment by sevlyar** [comment]
Fixed by #39
